ipc/provider/src/checkpoint.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+//! Bottom up checkpoint manager
+
+use crate::config::Subnet;
+use crate::manager::{BottomUpCheckpointRelayer, EthSubnetManager};
+use anyhow::{anyhow, Result};
+use fvm_shared::address::Address;
+use fvm_shared::clock::ChainEpoch;
+use ipc_identity::{EthKeyAddress, PersistentKeyStore};
+use std::fmt::{Display, Formatter};
+use std::sync::{Arc, RwLock};
+use std::time::Duration;
+
+/// Tracks the metadata required for both top down and bottom up checkpoint submissions, such as
+/// parent/child subnet and checkpoint period.
+pub struct CheckpointMetadata {
```

Comment:
It is really checkpoint metadata or the config for the bottom-up checkpoint?
```suggestion
pub struct CheckpointConfig {
```